python/ray/data/_internal/plan.py

```diff
@@ -449,12 +449,11 @@ def execute_to_iterator(
             )
 
         from ray.data._internal.execution.streaming_executor import StreamingExecutor
-        from ray.data._internal.execution.interfaces import ExecutionOptions
         from ray.data._internal.execution.legacy_compat import (
             execute_to_legacy_block_iterator,
         )
 
-        executor = StreamingExecutor(ExecutionOptions())
+        executor = StreamingExecutor(copy.deepcopy(ctx.execution_options))
```

Comment:
given `execution_options` could be muted in `StreamingExecutor`/`BulkExecutor`, shall we just provide a method in `DatasetContext.get_execution_options()` to just return a deep copy of `ctx.execution_options`?

Reply:
Missed this comment--- I'll file a followup.